File: software/examples/php/ExampleSimpleCurrent.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletIndustrialAnalogOutV2.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletIndustrialAnalogOutV2;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Industrial Analog Out Bricklet 2.0

$ipcon = new IPConnection(); // Create IP connection
$iao = new BrickletIndustrialAnalogOutV2(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Set output current to 4.5mA
$iao->setCurrent(4500);
$iao->setEnabled(TRUE);

echo "Press key to exit\n";
fgetc(fopen('php://stdin', 'r'));
$iao->setEnabled(FALSE);
$ipcon->disconnect();

?>

File: software/examples/rust/example_simple_current.rs
use std::{error::Error, io};

use tinkerforge::{industrial_analog_out_v2_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Industrial Analog Out Bricklet 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let iao = IndustrialAnalogOutV2Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Set output current to 4.5mA
    iao.set_current(4500).recv()?;
    iao.set_enabled(true).recv()?;

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;

    iao.set_enabled(false).recv()?;

    ipcon.disconnect();
    Ok(())
}
